### PR TITLE
js_pre_oop

### PR DESCRIPTION
**Pull request by Eugene50**
###A Tiny JS World

[Demo](https://eugene50.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/Eugene50/a-tiny-JS-world)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by Eugene50** [comment]
I`m correcting code!

File: submissions/Eugene50/a-tiny-JS-world/index.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Citizens {
+    constructor (species, gender, name, legs = '', hands = '', say){
```
Comment (OleksiyRudenko) [inline submissions/Eugene50/a-tiny-JS-world/index.js]: 1. Default values before parameters that do not have default values doesn't work properly.
You may want to reorder params.
2. Defaulting params that are numbers by nature with string values looks confusing.

Please, fix.

File: submissions/Eugene50/a-tiny-JS-world/index.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+     this.species = species;
+     this.gender = gender;
+     this.name = name;
+     this.legs = legs;
+     this.hands = hands;
+     this.say = say;   
+    }
+};
+
+let dog = new Citizens('dog','male','Sharik', 2, 4, 'Gaf!!!');
```
Comment (OleksiyRudenko) [inline submissions/Eugene50/a-tiny-JS-world/index.js]: A dog with 4 hands looks confusing. However, it is your world. Fixing is up to you.

File: submissions/Eugene50/a-tiny-JS-world/index.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+let cat = new Citizens('cat', 'male', 'Murzik', 4, 0, 'Mjau)' );
+let man = new Citizens('human','male','Alex', 2, 2, 'Hello to All!!!');
+let woman = new Citizens('human','female','Maria', 2, 2, 'I want coffee!!!');
+
+const all = [man, woman, cat, dog];
+all.forEach(element => {
+    print(`${element.species}; ${element.gender}; ${element.name}; ${element.legs}; ${element.hands}; ${element.say}`);
```
Comment (OleksiyRudenko) [inline submissions/Eugene50/a-tiny-JS-world/index.js]: You may form an array from element's properties and apply `.join`. E.g.
```
all.forEach(element => {
     print([element.species, element.gender, /* etc */ ].join("; "));
});
```

File: submissions/Eugene50/a-tiny-JS-world/index.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+})
```
Comment (OleksiyRudenko) [inline submissions/Eugene50/a-tiny-JS-world/index.js]: Please, tune your code editor so it ensures there is an empty line at the file end.

**Comment by Eugene50** [comment]
I have a question, why it is so important that  file must have the empty line at the end?

**Comment by OleksiyRudenko** [comment]
@Eugene50 On real job, when time resource is really restricted, code review is normally limited to the scope of changes only.
When you change a single character the entire string is marked changed.
When you have no newline at the end and add yet another line of code, the past last line is marked changed (as it now has a newline character) and new addition is marked as a change of code.
So, reviewer spends certain time to understand what has changed in the code and it frustrates when you realize that the only change in a certain line was a newline addition.
So it is a matter of caring about your collaborators and respect to their efforts and time.

File: submissions/Eugene50/a-tiny-JS-world/index.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+class Citizens {
+    constructor (species, gender, name, say, legs = 'null', hands = 'null' ){
```
Comment (OleksiyRudenko) [inline submissions/Eugene50/a-tiny-JS-world/index.js]: From past comment on the very same topic:
> 2. Defaulting params that are numbers by nature with string values looks confusing.
> Please, fix.

These are still strings. `null` and `'null'` are totally different things.
Please, also read about difference between `null` and `undefined` before you decide what to use.
You also may assume that legs are something defined as every citizen has legs.
Please, make this more natural.

**Pull request merged**